### PR TITLE
fix(s3): bugfixing the s3 endpoint parser

### PR DESCRIPTION
**Pull request by alexforsyth**
##### Checklist

- [x] `npm run test` passes
- [x] `.d.ts` file is updated
- [x] changelog is added, `npm run add-change`
- [x] run `bundle exec rake docs:api` and inspect `doc/latest/index.html` if documentation is changed
- [x] run `npm run integration` if integration test is changed
- [x] non-code related change (markdown/git settings etc)

### CODE REVIEW

File: dist/aws-sdk-react-native.js [hunk 1]
```diff
@@ -37216,14 +37217,25 @@ return /******/ (function(modules) { // webpackBootstrap
...
 	    var useArnRegion = req.service.config.s3UseArnRegion;
 
-	    endpoint.hostname = [
-	      accessPointArn.accessPoint + '-' + accessPointArn.accountId + outpostsSuffix,
-	      serviceName + dualStackSuffix,
-	      useArnRegion ? accessPointArn.region : req.service.config.region,
-	      dnsSuffix
-	    ].join('.');
-	    endpoint.host = endpoint.hostname;
+	    if (isObjectLambdaArn) {
```
Comment (AllanZhengYP) [inline dist/aws-sdk-react-native.js:37220]: Remove diffs in `dist/`, they are generated.

File: lib/services/s3.js [hunk 2]
```diff
@@ -350,14 +351,25 @@ AWS.util.update(AWS.S3.prototype, {
...
-    endpoint.host = endpoint.hostname;
+    if (isObjectLambdaArn) {
+      var serviceName = 's3-object-lambda';
+      // should be in the format: "accesspoint/${accesspointName}"
+      var accesspointName = accessPointArn.resource.split('/')[1];
+      endpoint.hostname = [
+        accesspointName + '-' + accessPointArn.accountId,
+        useArnRegion ? accessPointArn.region : req.service.config.region,
+        dnsSuffix
+      ].join('.');;
```
Comment (AllanZhengYP) [inline lib/services/s3.js:362]: nit.
```suggestion
      ].join('.');
```

Comment (alexforsyth) [reply lib/services/s3.js]: done

File: dist/aws-sdk-react-native.js (hunk 1 above)
Comment (alexforsyth) [reply dist/aws-sdk-react-native.js]: done

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**